ksqldb-engine/src/main/java/io/confluent/ksql/embedded/KsqlContext.java

```diff
@@ -74,13 +75,19 @@
    */
   public static KsqlContext create(
       final KsqlConfig ksqlConfig,
-      final ProcessingLogContext processingLogContext
+      final ProcessingLogContext processingLogContext,
+      final MetricCollectors metricCollectors
   ) {
     Objects.requireNonNull(ksqlConfig, "ksqlConfig cannot be null.");
     final ServiceContext serviceContext =
         ServiceContextFactory.create(ksqlConfig, DisabledKsqlClient::instance);
     final MutableFunctionRegistry functionRegistry = new InternalFunctionRegistry();
-    UserFunctionLoader.newInstance(ksqlConfig, functionRegistry, ".").load();
+    UserFunctionLoader.newInstance(
+        ksqlConfig,
+        functionRegistry,
+        ".",
+        metricCollectors.getMetrics()
```

Comment:
Rather than just throwing in the kitchen sink, I opted to only pass the Metrics instead of MetricCollectors through to the locations that just need the reduced scope.